x-pack/plugins/embeddable_enhanced/public/drilldowns/url_drilldown/url_drilldown_scope.tsx

```diff
@@ -0,0 +1,322 @@
+    return getEventScopeFromValueClickTriggerContext(eventScopeInput, deps, opts);
+  } else {
+    throw new Error("UrlDrilldown [getEventScope] can't build scope from not supported trigger");
+  }
+}
+
+async function getEventScopeFromRangeSelectTriggerContext(
+  eventScopeInput: RangeSelectContext
+): Promise<RangeSelectTriggerEventScope> {
+  const { table, column: columnIndex, range } = eventScopeInput.data;
+  const column = table.columns[columnIndex];
+  return cleanEmptyKeys({
+    key: toPrimitiveOrUndefined(column?.meta?.aggConfigParams?.field) as string | undefined,
+    from: toPrimitiveOrUndefined(range[0]) as string | number | undefined,
+    to: toPrimitiveOrUndefined(range[range.length - 1]) as string | number | undefined,
+  });
+}
+
+async function getEventScopeFromValueClickTriggerContext(
+  eventScopeInput: ValueClickContext,
+  deps: { getOpenModal: () => Promise<OverlayStart['openModal']> },
+  opts: { allowPrompts: boolean } = { allowPrompts: false }
+): Promise<ValueClickTriggerEventScope> {
+  const negate = eventScopeInput.data.negate ?? false;
+  const { table, column: columnIndex, value } = await getSingleValue(
+    eventScopeInput.data.data,
+    deps,
+    opts
+  );
+  const column = table.columns[columnIndex];
+  return cleanEmptyKeys({
+    key: toPrimitiveOrUndefined(column?.meta?.aggConfigParams?.field) as string | undefined,
+    value: toPrimitiveOrUndefined(value) as string | number | boolean,
+    negate,
+  });
+}
+
+/**
+ * @remarks
+ * Difference between `event` and `context` variables, is that real `context` variables are available during drilldown creation (e.g. embeddable panel)
+ * `event` variables are mapped from trigger context. Since there is no trigger context during drilldown creation, we have to provide some _mock_ variables for validating and previewing the URL
+ */
+export function getMockEventScope([trigger]: UrlTrigger[]): UrlDrilldownEventScope {
+  if (trigger === SELECT_RANGE_TRIGGER) {
+    return {
+      key: '__testKey__',
+      from: new Date(Date.now() - 15 * 60 * 1000).toISOString(), // 15 minutes ago
+      to: new Date().toISOString(),
+    };
+  } else {
+    return {
+      key: '__testKey__',
+      value: '__testValue__',
+      negate: false,
+    };
+  }
+}
+
+function toPrimitiveOrUndefined(v: unknown): string | number | boolean | undefined {
+  if (typeof v === 'number' || typeof v === 'boolean' || typeof v === 'string') return v;
+  if (typeof v === 'object' && v instanceof Date) return v.toISOString();
+  if (typeof v === 'undefined' || v === null) return undefined;
+  return String(v);
+}
+
+function cleanEmptyKeys<T extends Record<string, any>>(obj: T): T {
+  Object.keys(obj).forEach((key) => {
+    if (obj[key] === undefined) {
+      delete obj[key];
+    }
+  });
+  return obj;
+}
+
+/**
+ * VALUE_CLICK_TRIGGER could have multiple data points
+ * Prompt user which data point to use in a drilldown
+ */
+async function getSingleValue(
```

Comment:
@streamich, @ppisljar, this is an alternative to `getSingleValue` popup that we discussed: https://github.com/Dosant/kibana/pull/5
We can decide later if we want to go with `event.points` instead of popup